### PR TITLE
Linuxbuild

### PR DESCRIPTION
**Pull request by boomskats**
Adding support for building the examples on Linux x86_64. And probably some gratuitous cosmetics.

### CODE REVIEW

File: examples/OpenGL/Makefile [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
-	$(clang) $(includes) $(options) $(link) \
-	   -o opengl demo.cpp ../../wander.cpp
+	$(clang) $(includes) $(includesextra) \
+		$(options) $(optionsextra) \
+		$(link) $(linkextra) -o $(target) $(sources)
 
-run: build
-	./opengl
\ No newline at end of file
+clean:
```
Comment (seanisom) [inline examples/OpenGL/Makefile:47]: nit: can we add the `make run` target back? It is used in the demo

Comment (boomskats) [reply examples/OpenGL/Makefile:47]: Good catch - unintentional. Back in now

**Comment by seanisom** [comment]
@boomskats  - can you resolve the merge conflicts with the makefile? Will make sure OSX is still working after that, then merge

**Comment by seanisom** [comment]
Handled on this end - thanks for the contribution @boomskats!

**Pull request merged**